cli/src/semgrep/app/scans.py

```diff
@@ -325,7 +324,9 @@ def report_findings(
             else 0,
             "dependency_parser_errors": [e.to_json() for e in dependency_parser_errors],
             "stats": {
-                "findings": len(new_matches),
+                "findings": len(
+                    [match for match in new_matches if not match.is_prev_scan]
```

Comment:
This seems error prone and requires a higher mental load. Do the transient matches need to be reported in the same place as matches? aka can we partition the transient matches and actual matches earlier in the code and deal with them in a combined data structure as little as possible?

Reply:
Because of the way scan_source is annotated, you can still turn off this functionality even with cli side filtering by just not sending any extra annotated rules so I don't find the two way door argument compelling. And the downside of a field we expect but is never used seems minimal.

If we really want to avoid making app side changes I would still make changes on the cli side to have two separate lists for new_matches and transient_new_matches and if you really want to just use a single list in the API, combine them at the end. The error states I'm hoping to avoid are somehow printing, counting, blocking etc. on any new_match that is a is_prev_scan which is highly likely if we keep them all in the same object since now any time that list is used, the programmer needs to be aware that it needs to be filtered.

Reply:
Take for example line 323 above. If app sends the transient policy with associated actions, that might be wrong because we didn't filter out the transient rules

Reply:
Oooh okay yes, I do find the argument for programmer needing to have context around the prev scan stuff while making any changes to the rules or matches list compelling. I do see that being a pitfall for anyone not aware about these changes.

Okay great, I can separate out these findings upstream and keep it that way until we finally end up making an API call where we can merge it.

Thanks, @brendongo!

Reply:
@brendongo for line 323, it shouldn't still break anything as we annotated the "transient scan rules" in the backend to never be blocking. Which means those matches will never be included in line 323. However, yes, I do see that being confusing without devs having enough context.

Reply:
Discussed offline w @brendongo that it's okay to merge this in for now as we are going to port to OCaml soon.